### PR TITLE
Fix: Enable Expense author to edit invited payee email

### PR DESCRIPTION
**Pull request by kewitz**
Resolves https://github.com/opencollective/opencollective/issues/7080

### CODE REVIEW

File: server/graphql/common/expenses.ts [hunk 1]
```diff
@@ -1559,18 +1571,46 @@ export async function editExpenseDraft(req: express.Request, expenseData: Expens
...
+        inviteUrl,
+      })
+      .catch(e => {
+        logger.error('An error happened when creating the COLLECTIVE_EXPENSE_INVITE_DRAFTED activity', e);
+        reportErrorToSentry(e);
+      });
+    if (config.env === 'development') {
+      logger.info(`Expense Invite Link: ${inviteUrl}`);
+    }
+  }
```
Comment (Betree) [inline server/graphql/common/expenses.ts:1611]: Maybe we could have a common helper with `server/graphql/v2/mutation/ExpenseMutations.ts` to centralize this code

**Review by Betree — APPROVED** [review]
Lgtm!

File: server/graphql/common/expenses.ts [hunk 2]
```diff
@@ -1558,7 +1561,36 @@ const isValueChanging = (expense: Expense, expenseData: ExpenseData, key: string
...
+};
+
+export async function sendDraftExpenseInvite(
+  req: express.Request,
+  expense: Expense,
+  collective: Collective,
+  draftKey: string,
+): Promise<void> {
+  const inviteUrl = `${config.host.website}/${collective.slug}/expenses/${expense.id}?key=${draftKey}`;
+  expense
```
Comment (Betree) [inline server/graphql/common/expenses.ts:1579]: If we want the code that uses this function to be able to `await` on it

```suggestion
return expense
```

**Comment by kewitz** [comment]
Heads up @BenJam.

**Pull request merged**